File: TopInterview150/Stack/SimplifyPath/solution.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        output = ""
        current = ""
        stack = []
        path += "/"

        for c in path:
            if c == '/':
                if current == "..":
                    if len(stack) > 0:
                        stack.pop()
                elif current != "." and current != "":
                    stack.append(current)
                    print(current)
                current = ""
            else:
                current += c
        
        while len(stack) > 0:
            output = "/" + stack.pop() + output

        if output == "":
            return "/"
        return output

File: TopInterview150/Stack/SimplifyPath/solution.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stack<string> stack;
        string output = "";
        string current = "";
        path = path + "/";
        for (char c : path)
        {
            if (c == '/')
            {
                if (current == "..")
                { 
                    if (!stack.empty()) 
                    {
                        stack.pop();
                    }
                }
                else if (current == ".")
                    ;
                else if (current == "")
                    continue;
                else stack.push(current);
                current = "";
            }
            else current += c;
        }
        while (!stack.empty())
        {
            output = "/" + stack.top() + output;
            stack.pop();
        }
        if (output == "") return "/";
        return output;
    }
};